src/module/loader/module-resolver-cache.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include "module-resolver-cache.h"
+
+#include <mutex>
+#include <unordered_map>
+
+using namespace napa;
+using namespace napa::module;
+
+struct ModuleResolverCacheKey {
+    std::string _name;
+    std::string _path;
+
+    bool operator==(const ModuleResolverCacheKey& other) const {
+        return _name == other._name && _path == other._path;
+    }
+};
+
+namespace std {
+    template<>
+    struct hash<ModuleResolverCacheKey> {
+        std::size_t operator()(ModuleResolverCacheKey key) const {
+            return std::hash<std::string>{}(key._name) ^ std::hash<std::string>{}(key._path);
+        }
+    };
+}
+
+class ModuleResolverCache::ModuleResolverCacheImpl {
+public:
+    ModuleInfo Lookup(const char* name, const char* path);
+    void Insert(const char* name, const char* path, const ModuleInfo& moduleInfo);
+private:
+    std::mutex _lock;
+    std::unordered_map<ModuleResolverCacheKey, ModuleInfo> _resolvedModules;
```

Comment:
do we want LRU to avoid unlimited memory.

A simple implementation: https://leetcode.com/problems/lru-cache/discuss/ :-)